### PR TITLE
[CN-1206]: Add Notes to HA Documentation about Multi Member Routing

### PR DESCRIPTION
**Pull request by kutluhanmetin**

### CODE REVIEW

File: docs/modules/ROOT/pages/high-availability-mode.adoc
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/high-availability-mode.adoc:64]:
```suggestion
Subset routing is a client feature that allows a client to connect to a subset of members that is determined by a grouping strategy. To benefit from subset routing, you must set high availability mode, because this parameter is used to determine partition groups. Also, if the subset routing is used on client-side, `exposeExternally.type` must be set to `smart` in the Hazelcast CR.
```

File: docs/modules/ROOT/pages/high-availability-mode.adoc
Comment (rebekah-lawrence) [inline docs/modules/ROOT/pages/high-availability-mode.adoc:66]:
```suggestion
Example server and client configurations to run subset routing:
```

**Review by rebekah-lawrence — CHANGES_REQUESTED** [review]

**Review by rebekah-lawrence — COMMENTED** [review]
Looking good, but I'll hold off approving until the naming is confirmed (in case there is a bigger impact).

If you need the approval now, letme know.

File: docs/modules/ROOT/pages/high-availability-mode.adoc
Comment (JamesHazelcast) [inline docs/modules/ROOT/pages/high-availability-mode.adoc:62]:
```suggestion
== High Availability Mode and MULTI_MEMBER Routing
```

File: docs/modules/ROOT/pages/high-availability-mode.adoc
Comment (JamesHazelcast) [inline docs/modules/ROOT/pages/high-availability-mode.adoc:64]:
```suggestion
`MULTI_MEMBER` routing is a client feature that allows a client to connect to a subset of members that is determined by a grouping strategy. To benefit from `MULTI_MEMBER` routing, you must enable high availability mode, because this parameter is used to determine partition groups. Also, if the `MULTI_MEMBER` routing mode is used on the client-side, `exposeExternally.type` must be set to `smart` in the Hazelcast CR.
```

File: docs/modules/ROOT/pages/high-availability-mode.adoc
Comment (JamesHazelcast) [inline docs/modules/ROOT/pages/high-availability-mode.adoc:66]:
```suggestion
Example server and client configurations to use `MULTI_MEMBER` routing:
```

File: docs/modules/ROOT/pages/high-availability-mode.adoc
Comment (JamesHazelcast) [inline docs/modules/ROOT/pages/high-availability-mode.adoc:70]: this file name will need changing

File: docs/modules/ROOT/pages/high-availability-mode.adoc
Comment (JamesHazelcast) [inline docs/modules/ROOT/pages/high-availability-mode.adoc:98]:
```suggestion
    <network>
        <cluster-routing mode="MULTI_MEMBER">
          <grouping-strategy>PARTITION_GROUPS</grouping-strategy>
        </cluster-routing>
    </network>
```

File: docs/modules/ROOT/pages/high-availability-mode.adoc
Comment (JamesHazelcast) [inline docs/modules/ROOT/pages/high-availability-mode.adoc:112]:
```suggestion
  network:
    cluster-routing:
      mode: MULTI_MEMBER
      grouping-strategy: PARTITION_GROUPS
```

File: docs/modules/ROOT/pages/high-availability-mode.adoc
Comment (JamesHazelcast) [inline docs/modules/ROOT/pages/high-availability-mode.adoc:85]:
```suggestion
ClientConfig clientConfig = new ClientConfig();
ClientNetworkConfig networkConfig = clientConfig.getNetworkConfig();
networkConfig.getClusterRoutingConfig().setRoutingMode(RoutingMode.MULTI_MEMBER);
// PARTITION_GROUPS is the default strategy, so it does not need to be explicitly defined
networkConfig.getClusterRoutingConfig().setRoutingStrategy(RoutingStrategy.PARTITION_GROUPS);
```

**Review by JamesHazelcast — COMMENTED** [review]
New mode suggestions

**Review by rebekah-lawrence — APPROVED** [review]
LGTM

**Review by hasancelik — APPROVED** [review]

**Pull request merged**
